cloudferrylib/scheduler/scenario.py

```diff
@@ -117,3 +123,84 @@ def init_process_migrate(self, path):
         process = migrate['process']
         namespace = migrate['namespace']
         return process, namespace
+
+
+class ScenarioChecker():
+    def __init__(self, tasks, key, scenario):
+        self.tasks = tasks
+        self.key = key
+        self.scenario = scenario
+        self.missed_tasks = set()
+        self.duplicated_tasks = set()
+        self.missed_links = set()
+        self.links = set()
+        self.active_tasks = set()
+
+    def check(self):
+        self.check_dict_list(self.scenario)
+
+        # check missed links
+        for link in self.links:
+            if link not in self.active_tasks:
+                self.missed_links.add(link)
+
+        throw_exception = False
+        # log all errors if needed
+        if len(self.missed_links):
```

Comment:
Why do we need `len` function here? I guess `if self.missed_links:` will have the same effect.